clustering/infinispan/embedded/spi/src/main/java/org/wildfly/clustering/infinispan/marshall/DefaultEncoderRegistry.java

```diff
@@ -0,0 +1,163 @@
+import org.infinispan.commons.dataconversion.WrapperIds;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+
+/**
+ * Customer {@link EncoderRegistry} that supports transcoder removal.
+ * @author Paul Ferraro
+ */
+@Scope(Scopes.GLOBAL)
+public class DefaultEncoderRegistry implements EncoderRegistry {
+    @SuppressWarnings("deprecation")
+    private final Map<Short, org.infinispan.commons.dataconversion.Encoder> encoders = new ConcurrentHashMap<>();
+    private final Map<Byte, Wrapper> wrappers = new ConcurrentHashMap<>();
+    private final List<Transcoder> transcoders = Collections.synchronizedList(new ArrayList<>());
+    private final Map<MediaType, Map<MediaType, Transcoder>> transcoderCache = new ConcurrentHashMap<>();
+
+    @Override
+    public void registerTranscoder(Transcoder transcoder) {
+        this.transcoders.add(transcoder);
+    }
+
+    @Override
+    public Transcoder getTranscoder(MediaType fromType, MediaType toType) {
+        Transcoder transcoder = this.findTranscoder(fromType, toType);
+        if (transcoder == null) {
+            throw CONTAINER.cannotFindTranscoder(fromType, toType);
+        }
+        return transcoder;
+    }
+
+    @Override
+    public <T extends Transcoder> T getTranscoder(Class<T> targetClass) {
+        return targetClass.cast(this.transcoders.stream().filter(p -> p.getClass().equals(targetClass)).findAny().orElse(null));
+    }
+
+    @Override
+    public boolean isConversionSupported(MediaType fromType, MediaType toType) {
+        return fromType.match(toType) || this.findTranscoder(fromType, toType) != null;
+    }
+
+    @Override
+    public Object convert(Object object, MediaType fromType, MediaType toType) {
+        if (object == null) return null;
+        return this.getTranscoder(fromType, toType).transcode(object, fromType, toType);
+    }
+
+    @Override
+    public void unregisterTranscoder(MediaType type) {
+        synchronized (this.transcoders) {
+            Iterator<Transcoder> transcoders = this.transcoders.iterator();
+            while (transcoders.hasNext()) {
+                if (transcoders.next().getSupportedMediaTypes().contains(type)) {
+                    transcoders.remove();
+                }
+            }
+        }
+        this.transcoderCache.remove(type);
+        for (Map<MediaType, Transcoder> map : this.transcoderCache.values()) {
+            map.remove(type);
+        }
+    }
+
+    private Transcoder findTranscoder(MediaType fromType, MediaType toType) {
+        return this.transcoderCache.computeIfAbsent(fromType, mt -> new ConcurrentHashMap<>(4)).computeIfAbsent(toType, mt -> this.transcoders.stream().filter(t -> t.supportsConversion(fromType, toType)).findFirst().orElse(null));
+    }
+
+    @Deprecated
+    @Override
+    public org.infinispan.commons.dataconversion.Encoder getEncoder(Class<? extends org.infinispan.commons.dataconversion.Encoder> encoderClass, short encoderId) {
```

Comment:
Looks like a remnant that can just be an import here and other places (org.infinispan.commons.dataconversion.Encoder) unless it's intentional...

Reply:
This was intentional.  The Encoder interface and the methods of EncoderRegistry that use it are deprecated - so the FQCN was used to reduce the likelihood of merge conflicts when these methods are removed.